genesis-generator/main.hs

```diff
@@ -92,7 +113,11 @@ data InputConfig = InputConfig
   } deriving (Show, Generic)
 
 customOptions :: Options
-customOptions = defaultOptions { fieldLabelModifier = camelTo2 '_' }
+customOptions = defaultOptions { fieldLabelModifier = modifyFieldLabel }
+
+modifyFieldLabel :: String -> String
+modifyFieldLabel "rewardConstant" = "constant"
```

Comment:
this can also be solved by not using the `genericToJSON` function

for example:
https://github.com/input-output-hk/iohk-ops/blob/d96f4df93d42ca476adaca53e32f42f71fb7a6fa/iohk/common/Arch.hs#L90-L93
`.=` takes a key/value pair, and wraps it with something
`Aeson.object` then takes a list of those wrapped key/value pairs, and turns it into an aeson `Value`

then you can define your own mappings between haskell and json

https://github.com/input-output-hk/iohk-ops/blob/603e6b27c3cf79ee8cfefe31a2d8b516086e56c4/iohk/common/Nix.hs#L48-L52
a second simpler example